src/callbacks_step/stepsize_dg2d.jl

```diff
@@ -67,3 +67,53 @@ function max_dt(u::AbstractArray{<:Any,4}, t, mesh::ParallelTreeMesh{2},
 
   return dt
 end
+
+
+function max_dt(u::AbstractArray{<:Any,4}, t, mesh::StructuredMesh,
+                constant_speed::Val{false}, equations, dg::DG, cache)
+  # to avoid a division by zero if the speed vanishes everywhere,
+  # e.g. for steady-state linear advection
+  max_scaled_speed = nextfloat(zero(t))
+
+  @unpack size, coordinates_min, coordinates_max = mesh
+  dx = (coordinates_max[1] - coordinates_min[1]) / size[1]
+  dy = (coordinates_max[2] - coordinates_min[2]) / size[2]
+
+  for element in eachelement(dg, cache)
+    max_λ1 = max_λ2 = zero(max_scaled_speed)
+    for j in eachnode(dg), i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, j, element)
+      λ1, λ2 = max_abs_speeds(u_node, equations)
+      λ1 *= 0.5 * dy
+      λ2 *= 0.5 * dx
```

Comment:
We use a different Jacobian than you did. In a quadratic Cartesian grid, you used `dx/2` as Jacobian, so you could just use the physical fluxes without any further transformation to the reference element.
We may not have `dx = dy` now, so we used `dx*dy/4` as Jacobian (which is the actual Jacobian determinant computed with the metric terms of the transformation). Then, we had to use the transformed fluxes with the metric terms (which are just the constants `dx/2` and `dy/2` in this case) included.

In fact, we could remove the whole element loop since the Jacobian is the same for every element right now.

Reply:
Gotcha :+1: Don't remove anything, since you'll need it for the bilinear/curvilinear meshes anyways